common/web/gesture-recognizer/src/engine/headless/gesturePath.ts

```diff
@@ -138,4 +138,13 @@ export class GesturePath<Type, StateToken = any> extends EventEmitter<EventMap<T
 
     this.removeAllListeners();
   }
+
+  public toJSON(): any {
```

Comment:
This is not quite the right name -- because it's not returning JSON, it's returning a Javascript object.

Reply:
It's a special name and is correct.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

> For other objects: 
> - If the value has a toJSON() method, it's responsible to define what data will be serialized. Instead of the object being serialized, the value returned by the toJSON() method when called will be serialized. JSON.stringify() calls toJSON with one parameter, the key, which has the same semantic as the key parameter of the [replacer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter) function:
>
>    - if this object is a property value, the property name
>    - if it is in an array, the index in the array, as a string
>    - if JSON.stringify() was directly called on this object, an empty string